platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/migratable/ProfileMigratable.java

```diff
@@ -0,0 +1,244 @@
+
+  @VisibleForTesting static final ObjectMapper MAPPER = JsonFactory.create();
+
+  private final ApplicationManager appMgr;
+
+  private final FeatureManager featureMgr;
+
+  private final BundleManager bundleMgr;
+
+  @VisibleForTesting
+  ProfileMigratable(ApplicationManager appMgr, FeatureManager featureMgr, BundleManager bundleMgr) {
+    this.appMgr = appMgr;
+    this.featureMgr = featureMgr;
+    this.bundleMgr = bundleMgr;
+  }
+
+  /**
+   * Instantiates a profile migratable with the specified application and features services.
+   *
+   * @param appService the application services to use
+   * @param featuresService the features service to use
+   * @throws IllegalArgumentException if <code>applicationService</code> or <code>featuresService
+   * </code> is <code>null</code>
+   */
+  public ProfileMigratable(ApplicationService appService, FeaturesService featuresService) {
+    this(
+        new ApplicationManager(appService),
+        new FeatureManager(featuresService),
+        new BundleManager());
+  }
+
+  @Override
+  public String getVersion() {
+    return ProfileMigratable.VERSION;
+  }
+
+  @Override
+  public String getId() {
+    return "ddf.profile";
+  }
+
+  @Override
+  public String getTitle() {
+    return "Profile Migratable";
+  }
+
+  @Override
+  public String getDescription() {
+    return "Exports the state of applications, features, and bundles";
+  }
+
+  @Override
+  public String getOrganization() {
+    return "Codice";
+  }
+
+  @Override
+  public void doExport(ExportMigrationContext context) {
+    context
+        .getEntry(ProfileMigratable.PROFILE_PATH)
+        .store(
+            (r, os) ->
+                ProfileMigratable.MAPPER.writeValue(
+                    os,
+                    ImmutableMap.of( //
+                        ProfileMigratable.PROFILE_APPS,
+                        appMgr.exportApplications(),
+                        ProfileMigratable.PROFILE_FEATURES,
+                        featureMgr.exportFeatures(),
+                        ProfileMigratable.PROFILE_BUNDLES,
+                        bundleMgr.exportBundles())));
+  }
+
+  @Override
+  public void doImport(ImportMigrationContext context) {
+    try {
+      context.getEntry(ProfileMigratable.PROFILE_PATH).restore(this::restore);
+    } catch (MigrationException e) { // don't want to abort the import; just record the error
+      context.getReport().record(e);
+    }
+  }
+
+  @SuppressWarnings(
+      "squid:S2093" /* try-with-resource will throw IOException with InputStream and we do not care to get that exception */)
+  private boolean restore(MigrationReport report, Optional<InputStream> ois) throws IOException {
+    final InputStream is = ois.orElse(null);
+
+    if (is == null) {
+      throw new MigrationException("Import error: missing exported profile information.");
+    }
+    final Map<String, Object> pinfo;
+
+    try {
+      pinfo =
+          ProfileMigratable.MAPPER
+              .parser()
```

Comment:
❓ Will this be a problem if different systems have different default charsets?  Should we just use UTF-8 instead?

Reply:
🤔 It would seem that would be a problem. The same exist in the migration framework. Will address both.